### PR TITLE
Memory leak fix for multiple rocPRIM unit  tests

### PR DESCRIPTION
**Pull request by NguyenNhuDi**
Adding missing hip free calls and fixing hip graph calls.

### CODE REVIEW

**Comment by NguyenNhuDi** [comment]
This is not the completed PR. I still need to add more fixes as I go through the tickets

**Review by stanleytsang-amd — COMMENTED** [review]
@NguyenNhuDi Please look into the test failures in CI - they are global across all pipelines and are in files modified by your PR

**Comment by NguyenNhuDi** [comment]
PR still WIP, some sort of sync errors are causing CI failures

**Comment by NguyenNhuDi** [comment]
PR is done, waiting for ci tests and aproval

File: test/rocprim/test_utils_hipgraphs.hpp [hunk 1]
```diff
@@ -27,65 +28,74 @@
...
+
+            inline void endStreamCapture(hipStream_t & stream){
+                HIP_CHECK(hipStreamEndCapture(stream, &graph));
+            }
+            
+            inline void createGraph(){
+                HIP_CHECK(hipGraphInstantiate(&graph_instance, graph, nullptr, nullptr, 0));
+            }
+
+            inline void launchGraph(hipStream_t & stream, const bool sync=false){
```
Comment (umfranzw) [inline test/rocprim/test_utils_hipgraphs.hpp:52]: I think this function is the same as launchGraphHelper, below. We could probably remove one of them.

**Review by umfranzw — CHANGES_REQUESTED** [review]
Thanks for cleaning up the graph tests. I just have one small comment below.

File: test/rocprim/test_utils_hipgraphs.hpp (hunk 1 above)
Comment (NguyenNhuDi) [reply test/rocprim/test_utils_hipgraphs.hpp:52]: Done :D

**Review by umfranzw — APPROVED** [review]
I think this looks good now.

**Pull request merged**